applications/pelion_client/src/modules/net_ot.c

```diff
@@ -122,10 +123,24 @@ static void send_net_state_event(enum net_state state)
 
 static void set_net_state(enum net_state state)
 {
-	if (state != net_state) {
-		net_state = state;
-		send_net_state_event(net_state);
+	if (state == net_state) {
+		return;
+	}
+
+	net_state = state;
+
+	if (net_state == NET_STATE_CONNECTED) {
```

Comment:
it look as this was wrongly rebased (or needs rebasing) - operations for workqueue were moved
let's put the power_level restrictions to the new location

Reply:
alternatively you can do as for net_lte - place the code with the call to `set_net_state`

Reply:
Calls to workqueue removed here.

Reply:
Maybe the following to avoid IS_ENABLED repetition:
```
if (IS_ENABLED(CONFIG_PELION_CLIENT_POWER_MANAGER_EVENTS)) {
   power_manager_restrict(MODULE_IDX(MODULE), (net_state == NET_STATE_CONNECTED)?(POWER_MANAGER_LEVEL_ALIVE):(POWER_MANAGER_LEVEL_MAX));
}
```